UnitTests/Sources/UsersProvider/UsersProviderTests.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+@testable import ElementX
+
+@MainActor
+class SearchProviderTest: XCTestCase {
+    var provider: UsersProvider!
+    var clientProxy: MockClientProxy!
+    
+    override func setUpWithError() throws {
+        clientProxy = .init(userID: "")
+        provider = UsersProvider(clientProxy: clientProxy)
+    }
+    
+    func testQueryShowingResults() async throws {
+        clientProxy.searchUsersResult = .success(.init(results: [UserProfile.mockAlice], limited: true))
+        
+        let results = await (try? search(query: "AAA").get()) ?? []
```

Comment:
So I thought I hadn't seen this before. `.get()` was [deprecated](https://developer.apple.com/documentation/swift/task/get()-4i2gt) and replaced by `.value`. Would you mind having a search for all of these and replacing them with `.value` please?
```suggestion
        let results = await (try? search(query: "AAA").value) ?? []
```

Reply:
I'm calling .get() on Result<> and should be the right one

Reply:
Derp, sorry, my bad.